tests/test_bonfire.py

```diff
@@ -164,3 +165,79 @@ def test_ns_list_option_mine(mocker, caplog, namespace_list: list, reservation_l
     assert " ".join(["namespace-3", "false", "ready", "none"]) not in actual
     assert " ".join(["namespace-4", "false", "ready", "none"]) not in actual
     assert " ".join(["namespace-5", "true", "false", "none", "user-5"]) not in actual
+
+
+def test_ns_list_flag_output(
+    mocker,
+    caplog,
+    namespace_list: list,
+    reservation_list: list,
+):
+    caplog.set_level(100000)
+
+    mocker.patch("bonfire.bonfire.has_ns_operator", return_value=True)
+    mocker.patch("bonfire.namespaces.get_all_namespaces", return_value=namespace_list)
+    mocker.patch("bonfire.namespaces.get_json", return_value={})
+    mocker.patch("bonfire.namespaces.get_all_reservations", return_value=reservation_list)
+    mocker.patch("bonfire.namespaces.on_k8s", return_value=False)
+    mocker.patch("bonfire.namespaces.whoami", return_value="user-1")
+    mocker.patch("bonfire.openshift.process_template", return_value={})
+
+    runner = CliRunner()
+    result = runner.invoke(bonfire.namespace, ["list", "--output", "json"])
+
+    actual_ns_1 = json.loads(result.output).get("namespace-1")
+    actual_ns_2 = json.loads(result.output).get("namespace-2")
+    actual_ns_3 = json.loads(result.output).get("namespace-3")
+    actual_ns_4 = json.loads(result.output).get("namespace-4")
+    actual_ns_5 = json.loads(result.output).get("namespace-5")
+
+    assert {
```

Comment:
I might be wrong, but I think these asserts are just comparing length?

What about something more along the lines of:
```
d1 = {"hello": "hello", "hi": "hi", "bye": "bye"}

d2 = {"hi": "hi"}

assert all([item in d1.items() for item in d2.items()])
```

Reply:
Looks like `.items()` might be a set-like object based on what I read in https://docs.python.org/3/library/stdtypes.html#dict-views ... But its still kind of unclear. I think the above is probably more explicit and less confusing.

Reply:
Just updated with this method. One thing I had to do was delete the last key/value that was returned from click because it is the expires_in data that will have a changing value each time I run the test. Once I was able to do that I could test that the other three key/value pairs were present.